core/src/main/java/io/grpc/transport/ClientTransport.java

```diff
@@ -69,6 +71,15 @@ ClientStream newStream(MethodDescriptor<?, ?> method,
   void start(Listener listener);
 
   /**
+   * Pings the remote endpoint to verify that the transport is still active. When an acknowledgement
+   * is received, the given callback will be invoked using the given executor.
+   *
+   * <p>Transports that do not have any mechanism by which to ping the remote endpoint should
```

Comment:
I think instead such transports should either throw UnsupportedOperationException (or maybe fail the callback immediately). It is useful to know that the ping time provided is meaningful if the transport doesn't support pings.

For an in-process transport, it could still return 0, though.